src/pages/HistoryPage.js

```diff
@@ -28,18 +29,82 @@ const fakeHistories = [
 
 const HistoryPage = props => {
   const [msgType, setMsgType] = useState('give');
-  const histories = fakeHistories; // This should be retrieved on page load
+  const [gotTransactions, setGotTransactions] = useState(false);
+  const [transactions, setTransactions] = useState([]);
+
+  // const histories = fakeHistories; // This should be retrieved on page load
+  console.log('got?', gotTransactions);
+
+  if (!gotTransactions) {
+    props.firebase
+      .transactions()
+      .get()
+      .then(querySnapshot => {
+        const transacs = querySnapshot.docs.map(doc => {
+          return {id: doc.id, ...doc.data()};
+        });
+        transacs.forEach(transac => {
```

Comment:
Using ```map()``` is probably faster here.

Reply:
map won't change the original data which is my intent.